instagram/src/components/SearchBar/SearchBarContainer.js

```diff
@@ -0,0 +1,282 @@
+    height: 36px;
+    padding: 14px 0 0;
+    display: flex;
+    vertical-align: baseline;
+`;
+
+const LogoImageCam = styled.img`
+    width: 24px;
+    height: 24px;
+`;
+
+const LogoImageLogo = styled.img`
+    width: 103px;
+    height: 29px;
+`;
+
+const Divider = styled.span`
+    width: .5px;
+    height: 24px;
+    margin: 0 16px 0;
+    background: #3333;
+`;
+
+const InputWrapper = styled.div`
+    width: 215px;
+    height: 24px;
+    padding: 3px 10px 3px 23px;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    background-color: #fafafa;
+    border: 1px solid #dbdbdb;
+    border-radius: 3px;
+`;
+
+const SpanWrapper = styled.button`
+    width: 200px;
+    height: 28px;
+    padding: 7px 7px 7px 0;
+    position: absolute;
+    z-index: 3;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    border: none;
+    outline: 0;
+    text-decoration: none;
+    background-color: #fafafa;
+    visbility: visible;
+    ::focus {
+        visbility: hidden;
+    }
+`;
+
+const SearchSpan = styled.span`
+    height: 14px;
+    margin: 0 3px;
+    align-items: baseline;
+    color: #e3e3e3;
+    font-size: 14px;
+    text-color: #999;
+`;
+
+const SearchIcon = styled.img.attrs({
+    alt: "Instagram search icon",
+    src: `${IGSearchIcon}`
+})`
+    width: 10px;
+    height: 10px;
+    opacity: 0.3;
+`;
+
+const SearchInput = styled.input.attrs({
+    type: "text",
+    placeholder: "Search",
+})`
+    width: 177px;
+    height: 20px;
+    display: flex;
+    border: none;
+    background-color: #fafafa;
+    text-color: none;
+    ::focus {
+        outline: 0;
+        text-color: #999;
+    }
+`;
+
+const IconsWrapper = styled.div`
+    width: 377.5px;
+    height: 24px;
+    display: flex;
+    justify-content: flex-end;
+    align-items: center;
+`;
+
+const IconInnerWrapper = styled.div`
+    width: 132px;
+    height: 24px;
+    padding-left: 24px;
+`;
+
+const Icon1 = styled.img`
+    width: 24px;
+    height: 24px;
+    margin-right: 30px
+`;
+
+const Icon2 = styled.img`
+    width: 24px;
+    height: 24px;
+`;
+
+const Icon3 = styled.img`
+    width: 24px;
+    height: 24px;
+    margin-left: 30px
+`;
+
+export default class SearchBarContainer extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            isHidden: true
+        }
+    }
+
+    toggleHandler() {
+        this.setState({
+            isHidden: !this.state.isHidden
+        })
+    }
+    
+    render() {
+        return(
+            <Header>
+                <HeaderBox>
+                    <LogoHeader>
+                        <LogoImageCam alt="Instagram camera logo" src={CameraLogo} />
+                        <Divider />
+                        <LogoImageLogo alt="Instagram logo" src={IGLogo} />
+                    </LogoHeader>
+                    <InputWrapper>
+                        <SpanWrapper onClick={this.toggleHandler.bind(this)} >
```

Comment:
You can avoid having to bind `this` by making your toggleHandler method an arrow function.